api/server.spec.js

```diff
@@ -0,0 +1,123 @@
+const request = require("supertest");
+
+const server = require("./server.js");
+const db = require("../data/dbConfig");
+
+beforeEach(async () => {
+  await db("movies").truncate(); // reset the database before test
+});
+
+describe("GET /", () => {
+  it("should return status 200, json message Welcome to Movies API testing app", async () => {
+    const res = await request(server).get("/");
+
+    expect(res.status).toBe(200);
+    expect(res.type).toBe("application/json");
+    expect(res.body).toBe("Welcome to Movies API Testing App");
+  });
+});
+
+describe("GET /movies", () => {
+  beforeEach(async () => {
+    await db("movies").truncate(); // reset the database before test
+  });
+
+  it("should return status 200, json empty array", async () => {
+    const res = await request(server).get("/movies");
+
+    expect(res.status).toBe(200);
+    expect(res.type).toBe("application/json");
+    expect(res.body).toBeInstanceOf(Array);
+    expect(res.body.length).toBe(0);
+  });
+
+  it("should return an array of movies, if database has data", async () => {
+    await db("movies").insert({
+      title: "Shazam!",
+      description: "A boy is given the ability to become an adult superhero in times of need with a single magic word."
+    });
+    await db("movies").insert({
+      title: "Captain Marvel",
+      description:
+        "The story follows Carol Danvers as she becomes one of the universe’s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races. Set in the 1990s, Captain Marvel is an all-new adventure from a previously unseen period in the history of the Marvel Cinematic Universe."
+    });
+
+    const res = await request(server).get("/movies");
+    expect(res.status).toBe(200);
+    expect(res.type).toBe("application/json");
+    expect(res.body).toBeInstanceOf(Array);
+    expect(res.body.length).toBe(2);
+  });
+});
+
+describe("POST /movies", () => {
+  beforeEach(async () => {
+    await db("movies").truncate(); // reset the database before test
+  });
+
+  it("should add movie to the database", async () => {
+    const movieData = {
+      title: "Shazam!",
+      description: "A boy is given the ability to become an adult superhero in times of need with a single magic word."
+    };
+    const res = await request(server)
+      .post("/movies")
+      .send(movieData)
+      .set("Accept", "application/json");
+    expect(res.status).toBe(201);
+    expect(res.type).toBe("application/json");
+
+    const movie = await db("movies")
+      .where({ id: res.body })
+      .first();
```

Comment:
As the instructor said during lecture, you probably don't want to check the DB to see if the data was properly inserted. If the response returns the proper data then you can assume it worked